api/handlers/containerhandler.py

```diff
@@ -480,6 +481,37 @@ def get_groups_with_project(self):
         group_ids = list(set((p['group'] for p in self.get_all('projects'))))
         return list(config.db.groups.find({'_id': {'$in': group_ids}}, ['name']))
 
+    def set_project_template(self, **kwargs):
+        project_id = kwargs.pop('cid')
+        self.config = self.container_handler_configurations['projects']
+        self.storage = self.config['storage']
+        container = self._get_container(project_id)
+
+        template = self.request.json_body
+        validators.validate_data(template, 'project-template.json', 'input', 'POST')
+        payload = {'template': json.dumps(template)}
+        payload['modified'] = datetime.datetime.utcnow()
+
+        permchecker = self._get_permchecker(container)
+        result = permchecker(self.storage.exec_op)('PUT', _id=project_id, payload=payload)
+
+        if result.modified_count == 1:
+            sessions = self.storage.get_children(project_id, projection={'_id':1})
+            log.debug('sessions are {}'.format(sessions))
+            session_storage = self.container_handler_configurations['sessions']['storage']
+            for s in sessions:
+                log.debug('update session {}'.format(s))
+                session_storage.exec_op('PUT', s['_id'], payload={'project_has_template': True})
+            return {'modified': result.modified_count}
```

Comment:
It makes sense why this logic has to be in the handler, but could you add a code comment explaining why?